Doc/library/argparse.rst

```diff
@@ -2162,6 +2162,46 @@ Intermixed parsing
 
    .. versionadded:: 3.7
 
+Registering custom types or actions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. method:: ArgumentParser.register(registry_name, value, object)
+
+   It is possible to register custom actions, types, or other objects with
+   a parser using :meth:`!register`. This method is called when the action is
+   added to the parser. :meth:`!register` takes in the following arguments:
+
+   * *registry_name* - the name of the internal registry where the object
+     will be stored (e.g. ``action``, ``type``).
+
+   * *value* - the key under which the object will be registered.
+
+   * *object* - the callable object to be registered.
+
+   The following example shows how to register a custom type with a parser::
+
+      >>> import argparse
+      >>> import enum
+      >>> class WinterMonths(enum.StrEnum):
+      ...     JAN = 'January'
+      ...     FEB = 'February'
+      ...     DEC = 'December'
+      ...
+      >>> WinterMonths.__name__ = 'Winter Month'
+      >>> parser = argparse.ArgumentParser()
+      >>> parser.register('type', 'Winter Month', WinterMonths)
+      >>> parser.add_argument('--month', type=WinterMonths, action='store')
+      _StoreAction(option_strings=['--month'], dest='month', nargs=None, const=None, default=None, type=<enum 'Winter Month'>, choices=None, required=False, help=None, metavar=None, deprecated=False)
+      >>> parser.parse_args(['--month','January'])
+      Namespace(month=<Winter Month.JAN: 'January'>)
+      >>> parser.parse_args(['--month','March'])
+      usage: python.exe -m _pyrepl [-h] [--month MONTH]
+      python.exe -m _pyrepl: error: argument --month: invalid Winter Month value: 'March'
```

Comment:
This pyrepl stuff is confusing. Perhaps it would be better to formulate the example as a script and then call it, which is after all how arparse is used?

Reply:
Yeah, I had this in a script block but that style isn't really used elsewhere in this doc so I reformatted. I agree though, that'd be better. I'll reformat again and then maybe one of our docs experts can weigh in.

Reply:
How about changing the program name?